### ISSUE TITLE
Error: Line number (37) greater than maximum line 36

### ISSUE THREAD

**Issue by vjancik** [AtomLinter/linter-eslint#496]
```
Error: Line number (37) greater than maximum line (36)
    at Object.rangeFromLineNumber (/home/vix/.atom/packages/linter-eslint/node_modules/atom-linter/lib/index.js:179:11)
    at /home/vix/.atom/packages/linter-eslint/lib/main.js:172:33
    at Array.map (native)
    at /home/vix/.atom/packages/linter-eslint/lib/main.js:155:27
```

I previously got the error with **Line number (11) greater than maximum line (10)** so I suspect it's some off by one error but I can't be sure.

**Comment by Arcanemagus** [comment]
Are you able to share the file _exactly_ as it was when this triggered, as well as the configuration that causes this to show up?

**Comment by vjancik** [comment]
This is my global `.eslintrc` I was using at the time.
```
{
  "extends" : "airbnb",
  "rules":{
        "no-console":0,
        "prefer-template":0,
        "strict":0,
        "func-names":0,
        "prefer-rest-params":0,
        "consistent-return":0,
        "no-param-reassign":0,
    }
}
```

Everything else is default. As for the version, I am using the latest.
And I can't pinpoint what exactly I  was doing, or rather, it didn't seem to occur at any particular event but rather at random. On error list update.

Wild guess: Could it have tried to recheck an error on the last line 37 but I deleted some line thus it failed?

**Comment by Arcanemagus** [comment]
That error message only happens when the linter (`eslint`) returns an invalid range back to us, in this case the line it gave back was 37, when the file you were working on only had 36 lines. Without the file in a state that triggers that error we can't really diagnose what the root cause of the error is, although I'm a bit surprised you hit one with that config as most of the time when `eslint` gives us an invalid range it's the fault of a buggy plugin.

If you see this again please _immediately_ save the file and share it here along with what you were editing in the file at that moment so we can diagnose where the error lies.

**Comment by IanVS** [comment]
I've seen it occasionally as well.  I recall it being transitory though, and as soon as I saved it would disappear.  I'll keep an eye out for it though.

**Comment by Arcanemagus** [comment]
I'm marking this as unable to reproduce and closing for now, if anyone has a file that is in a state that shows this issue please comment here or file a new issue!

**Comment by IanVS** [comment]
I just had this happen, but was unable to reproduce it a second time through undo/redo or repeating the edits I was making.  It seems to be some kind of race condition.

**Comment by Arcanemagus** [comment]
Hmmm, you could try running with a breakpoint set where that error is thrown? Might be some accessible information in the closures...
